### PR TITLE
Issue 106/deployment

### PR DESCRIPTION
**Pull request by nosnickid**

### CODE REVIEW

**Comment by nosnickid** [comment]
This closes #106

**Review by ajtucker — COMMENTED** [review]
Thanks for sorting this out and writing it up, @nosnickid .

The only bit I don't follow is how the `ADDONS` make it into the Docker build. For the "production" Docker image, we needed to customise things a bit (e.g. the `BUILD_TARGET=server` env to avoid the whole client side hotloading and two node servers thing). I think for the standard Volto Docker image, they use this `ADDONS` env file along with a custom `entrypoint.sh` to fiddle with the `package.json` when you spin up the container. However, for our own custom Dockerfile, we don't do that (yet).

Note, I can't see where the standard Volto `Dockerfile` is now. The closest is https://hub.docker.com/r/plone/volto/dockerfile, but I can't see it in the GitHub repo.

**Comment by nosnickid** [comment]
Ah, that linked Dockerfile is more about using `yo` to generate a baseline volto site, which we've come a long way from.

The "build" process for volto is a bit unusual, in that the built assets still need to access `node_modules` on load. It's like this because volto dynamically loads plugins from the package.json config. It's quite unusual really. 

I've updated `docs/volto.md` that explains it in a bit more detail. Let me know if it's still not clear.

File: docs/volto.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+The addons that volto loads are normally specified in the `package.json:addons` array. That can be
+overridden using the ADDONS env var.
+
+There are two aspects to using a volto addon. 
+
+First, the add on must be available as either a `node_module` or as a `mrs-developer` managed workspace in 
+this project. This step means the addon source will be present, and any dependencies it has will also be installed.
+
+Second, to use the addon, it must be present in the `package.json:addons` array. This will cause the volto to load
```
Comment (ajtucker) [inline docs/volto.md:30]: Ok, gotcha. I hadn't seen that this was possible and thought that there was some jiggery pokery going on in the Docker entry-point script to effectively update the `package.json`.

Have you got a link to the docs where they talk about this env variable overriding the `package.json:addons` array?

Comment (ajtucker) [reply docs/volto.md:30]: Found it! (Neat life hack -- I used Google :) )

https://docs.voltocms.com/configuration/environmentvariables/#use-add-ons-via-the-addons-environment-variable

Comment (nosnickid) [reply docs/volto.md:30]: Cunning ploy. Update looks good.

**Review by ajtucker — APPROVED** [review]
I've tried this out locally and it seems to work a treat, thanks @nosnickid.

**Pull request merged**